sorts/cyclic_sort.py

```diff
@@ -0,0 +1,52 @@
+"""
+This is a pure Python implementation of the Cyclic Sort algorithm.
+
+For doctests run following command:
+python -m doctest -v cyclic_sort.py
+or
+python3 -m doctest -v cyclic_sort.py
+For manual testing run:
+python cyclic_sort.py
+"""
+
+
+def cyclic_sort(nums: list) -> list:
+    """
+    Sorts the input list in-place using the Cyclic Sort algorithm.
```

Comment:
I'd recommend adding the input constraints.

This algorithm only works correctly when the elements range from `1` to `n` with no duplicates. Stating this constraint on the inputs to any potential users of the algorithm can help prevent confusion should they provide a list of integers that does not fit the criteria and end up in an infinite loop or some other unexpected behaviour.